### PR TITLE
es5 -> es6, node -> 8, callbacks -> promises, jshint -> eslint, drop lodash dep., audit -> 0 vuln.

### PR DESCRIPTION
**Pull request by koresar**
This `sqs-producer` is getting very popular year after year.

image

The node.js has moved to ES6 and promises a few years ago.

### I believe it's time to release version 2.0

This PR brings the long needed support of promises and async/await syntax.

Full list of changes:
* got rid of callbacks, only promises are supported now
* No logic changes. At all!
* same for the unit tests
* added `"engines": { "node": ">=8.0.0" }` to package.json
* removed node versions 4 and 6 from travis.yml, added 10 and 12
* replaced jshint with eslint for simplicity and better ES6 support
* Updated README with the v2.0 breaking changes
* Removed `lodash` as a dependency. It was used only once in the code base.
* Upgraded `nyc` dependency to kill all `npm audit` vulnerability warnings
* Updated `.jsbeautifyrc`, I avoided a breaking change the package had some time ago
* Updated `.npmignore` because coverage report [was published to NPM](https://unpkg.com/browse/sqs-producer@1.6.3/.nyc_output/)
* updated CONTRIBUTING.md with a tip how to run `js-beautify`

Now, how would I get this published @nspragg @niklasR ?

### CODE REVIEW

**Comment by jgcmarins** [comment]
https://twitter.com/jgcmarins/status/1230928879612809221

**Comment by chasingSublimity** [comment]
hiya folks! Are there plans for merging this PR? I'm considering using this package in a project but don't want to pull it in if there aren't any plans for continued features/maintenance. :)

Thanks!

**Comment by jgcmarins** [comment]
We've been using it in production, it works great.
But this PR should be merged soon.

**Comment by nspragg** [comment]
@koresar - Thank you for raising this PR. However, for future reference, it's recommended that you raise an issue first. This allows us to discuss and plan work accordingly.

File: lib/producer.js [hunk 1]
```diff
@@ -23,44 +19,53 @@ function Producer(options) {
...
 function entryId(entry) {
   return entry.Id;
 }
 
+function isString(s) {
+  return typeof s === 'string';
+}
+
+function isObject(s) {
+  return typeof s === 'object';
```
Comment (jeanrauwers) [inline lib/producer.js:37]: Similar here

```suggestion
  return s && typeof s === 'object' && s instanceof Object;
```

**Comment by koresar** [comment]
@nspragg sorry for rushing. I browsed a number of issues and saw a comment somewhere that there are plans to migrate to promises. That was my mistake. Apologies.

@jeanrauwers Review comments done.

**Comment by jeanrauwers** [comment]
> hiya folks! Are there plans for merging this PR? I'm considering using this package in a project but don't want to pull it in if there aren't any plans for continued features/maintenance. :)
> 
> Thanks!

We are working on it! thanks

**Comment by nspragg** [comment]
LGTM 👍

**Comment by koresar** [comment]
Thanks for your help gentlemen.

Should I now change it to be ES6 class? (in a new pr)

**Comment by nspragg** [comment]
@koresar - the prototype work is now in progress. Many thanks.

**Comment by jeanrauwers** [comment]
> Thanks for your help gentlemen.
> 
> Should I now change it to be ES6 class? (in a new pr)

There is some work going on at this moment on that, also we are looking to migrate to TS.

**Pull request merged**